plugins/index.ts

```diff
@@ -119,12 +119,9 @@ const deployOrUpgradeProxy = async <
     // )
   ) {
     hre.trace('Deploying proxy and instance', contractName);
-    const fireblocksSigner = signer as FireblocksSigner;
-    if (typeof fireblocksSigner.setNextTransactionMemo === 'function') {
-      fireblocksSigner.setNextTransactionMemo(
-        `Deploy proxy and instance for ${contractName}`
-      );
-    }
+    const fireblocksSigner = signer as unknown as FireblocksWeb3Provider;
```

Comment:
why is this casting needed? What is signer initially? Should we instead add a check on the type of the signer so we can use type narrowing and throw an error instead if the wrong signer is being used?

Reply:
same question for the several other places this happens